src/pages/guides/getting-started/editing-the-document/index.md

```diff
@@ -0,0 +1,120 @@
+    </footer>
+  </body>
+</html>
+```
+
+Note that we also changed the button's `id` to `"btnRename"`. With that (after re-loading, in case you don't have UDT's _"Watch"_ function enabled), your panel should look like this:
+
+![A Photoshop panel with a button "Rename Layers"](panel.png)
+
+## Step 2: Adjust the `index.js`
+
+Next, you need to adjust the `index.js` code to match the new functionality. You might have already tried (while tinkering around) to just use something like this:
+
+```js
+layer.name = `${layer.name} (${layer.opacity} %)`;
+```
+
+While that's already close, you will run into an error stating that "Such events are only allowed from inside the modal scope":
+
+![Chrome Developer Tools showing an error: "Uncaught Error: Event: select may modify the state of Photoshop. Such events are only allowed from inside a modal scope"](modal-execution-error.png)
+
+<InlineAlert variant="info" slots="text" />
+
+To open a Chrome Developer Tools window for your plugin, select _"Debug"_ in the plugin actions menu in the UXP
+Developer Tool.
+
+To get this to work, you need to wrap your code in a `window.require('photoshop').core.executeAsModal(targetFunction, options)` call. This creates a context in which your plugin has exclusive access to manipulating the document and thus ensures that no changes can be made to the document while your code executes (which could result in strange errors). You need to use `executeAsModal` for any operation that could edit the document. [Learn more about modal execution](/src/pages/ps_reference/media/executeasmodal.md).
+
+Adjust your `index.js` code to look like this:
+
+```js
+function renameLayerNames() {
+  return window.require("photoshop").core.executeAsModal(
+    () => {
+      // now, we have write-access
+
+      const app = window.require("photoshop").app;
+
+      app.activeDocument.layers.forEach((layer) => {
+        layer.name = `${layer.name} (${layer.opacity} %)`;
+      });
+    },
+    {
+      commandName: "Rename layers",
+    }
+  );
+}
+
+document
+  .getElementById("btnRename")
+  .addEventListener("click", renameLayerNames);
+```
+
+When you reload your plugin using the UDT and press the new button, you can see that the layer names get adjusted accordingly:
+
+![Photoshop Layers panel with layer names "Layer 1 (60 %)" and "Background (100 %)"](layer-names.png)
+
+## Step 3: Solve problem with consecutive executions
+
+If you already have some development experience (or pressed the _"Rename Layers"_ buttons more than one time), you might already see a problem with our implementation. While the button works perfectly for the first time, every other time, it adds another percentage to the layer name:
+
+![A layer with the name "Layer 1 (60 %) (60 %) (60 %)"](layer-name-problem.png)
+
+To fix this, change your layer loop to the following:
+
+```js
+// [...]
+app.activeDocument.layers.forEach((layer) => {
+  const baseName = layer.name.match(/^(.*?)( \(\d+ %\))?$/)[1];
```

Comment:
Wonder if this might be too much for some new developers? Is there an easier way to strip these off?

Reply:
I thought about this as well / don't really like it as it is.

I don't think there's any good way to achieve this without RegEx, unfortunately (it will only get more complicated by forcefully trying to avoid it), but I could maybe make it a bit clearer using a dedicated variable for the RegEx, a comment explaining the RegEx (including a link to a corresponding regex101 page), etc. 🤔

Reply:
I made it a bit more readable with the previous commit. Not sure if it's sufficient => feel free to re-review.